### PR TITLE
Ci Setup for pecl

### PR DESCRIPTION
**Pull request by ruudboon**
* Type: bug fix
* Link to issue: #14658 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:
Created some automation for Pecl package creation. Currently we don't have support for testing the package under windows. The package is manually verified and approved by the Pecl Windows team.

### CODE REVIEW

File: .github/workflows/pecl.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      - name: Download phalcon pecl package
+        uses: actions/download-artifact@v1
+        with:
+          name: phalcon-pecl
+      - name: Install package
+        run: |
+          export MAKEFLAGS="-j$(getconf _NPROCESSORS_ONLN)"
+          sudo pecl -v install phalcon-pecl/phalcon-pecl.tgz
+      - name: Verify install
+        run: php -m | grep phalcon
```
Comment (Jeckerson) [inline .github/workflows/pecl.yml]: Does action build fail in case grep return 1?

Comment (ruudboon) [reply .github/workflows/pecl.yml]: If Phalcon doesn't exists it fails indeed

Comment (ruudboon) [reply .github/workflows/pecl.yml]: @Jeckerson https://github.com/ruudboon/cphalcon/runs/366650215#step:6:1

Comment (Jeckerson) [reply .github/workflows/pecl.yml]: :+1:

**Review by Jeckerson — DISMISSED** [review]

**Review by niden — DISMISSED** [review]

File: package.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  <contents>
+    <dir name="/">
+      <file name="config.w32" role="src" />
+      <file name="phalcon.zep.c" role="src"  />
+      <file name="config.m4" role="src"  />
+      <file name="php_phalcon.h" role="src" />
+      <file name="phalcon.zep.h" role="src" />
+      <file name="LICENSE.txt" role="doc" />
+      <file name="CHANGELOG-4.0.md" role="doc" />
+      <file name="CODE_OF_CONDUCT.md" role="doc" />
```
Comment (sergeyklay) [inline package.xml]: Can you add CODE_OWNERS.TXT too? And FYI: PECL recognizes also CREDITS file in the following  format:
```
<short, one line description of the project>
<author 1>, <author 2>, <author N>
```

File: package.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    </dir>
+  </contents>
+  <dependencies>
+    <required>
+      <php>
+        <min>7.2.0</min>
```
Comment (sergeyklay) [inline package.xml]: Can you please add `max` too

File: .github/workflows/pecl.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      - 'package.xml'
+      - 'build/php7/safe/**'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build Phalcon Pecl Package
+    steps:
+      - name: Show system info
+        run: |
+          uname -m
```
Comment (sergeyklay) [inline .github/workflows/pecl.yml]:
```diff
-        run: |
+        run: uname -m
-          uname -m
```

File: .github/workflows/pecl.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+name: Pecl builder
+
+on:
+  create:
+    tags:
+  push:
+    paths:
+      - '.github/workflows/pecl.yml'
```
Comment (sergeyklay) [inline .github/workflows/pecl.yml]:
```suggestion
      - .github/workflows/pecl.yml
```

Use quotes `'` only to _force_ a string.

File: .github/workflows/pecl.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      - 'package.xml'
```
Comment (sergeyklay) [inline .github/workflows/pecl.yml]: the same

File: package.xml (hunk 3 above)
Comment (ruudboon) [reply package.xml]: Not sure. I didn't found a wildcard option for package.xml to do for example ^7.4 
I could add <max>8.0.0</max>?

File: package.xml [hunk 7]
```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<package packagerversion="1.10.5" version="2.0" xmlns="http://pear.php.net/dtd/package-2.0" xmlns:tasks="http://pear.php.net/dtd/tasks-1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pear.php.net/dtd/tasks-1.0 http://pear.php.net/dtd/tasks-1.0.xsd http://pear.php.net/dtd/package-2.0 http://pear.php.net/dtd/package-2.0.xsd">
+  <name>phalcon</name>
+  <channel>pecl.php.net</channel>
+  <summary>Phalcon is a full stack PHP framework offering low resource consumption and high performance.</summary>
+  <description>
+    Phalcon is an open source full stack framework for PHP, written as a C-extension. Phalcon is optimized for high performance. Its unique architecture allows the framework to always be memory resident, offering its functionality whenever its needed, without expensive file stats and file reads that traditional PHP frameworks employ.</description>
+  <lead>
```
Comment (sergeyklay) [inline package.xml]: Could you add all team members?

Comment (ruudboon) [reply package.xml]: I think it should match up with the pecl account. Will check this otherwise I will add you guys

File: package.xml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    ## Fixed
+    - Fixed `Phalcon\Mvc\Model\Criteria` Di isn't set when using `Criteria::fromInput()` [#14538](https://github.com/phalcon/cphalcon/issues/14639)
+    - Fixed `Phalcon\Db\Dialect\Mysql` removing unnecessary parentheses for `double` and `float` [#14645](https://github.com/phalcon/cphalcon/pull/14645) [@pfz](https://github.com/pfz)
+    - Fixed `Phalcon\Http\Cookie::delete` to parse the correct parameters - cannot use alternative syntax until PHP 7.3 [#14643](https://github.com/phalcon/cphalcon/issues/14643)
+    - Fixed `Phalcon\Mvc\Model::__isset` to take into account non visible properties by checking the getter if it exists [#13518](https://github.com/phalcon/cphalcon/issues/13518) [#13900](https://github.com/phalcon/cphalcon/issues/13900)
+    - Fixed `Phalcon\Mvc\Model::__set` to return a more informative message if we are tying to access a non visible property [#13518](https://github.com/phalcon/cphalcon/issues/13518) [#13900](https://github.com/phalcon/cphalcon/issues/13900)
+  </notes>
+  <contents>
+    <dir name="/">
+      <file name="config.w32" role="src" />
```
Comment (sergeyklay) [inline package.xml]: AFAIK the right format is
```
      <file md5sum="..." name="..." role="..." />
```

I would like to provide md5sum for every file.

Comment (ruudboon) [reply package.xml]: This is generated by the pecl cmd. This is a template. Final output will contain this

File: package.xml (hunk 3 above)
Comment (sergeyklay) [reply package.xml]: Just 8.0.0

File: package.xml (hunk 8 above)
Comment (ruudboon) [reply package.xml]: @sergeyklay btw. I found this out the hard way. Implemented an md5 generator at first 😅

File: package.xml (hunk 7 above)
Comment (ruudboon) [reply package.xml]: Need to dismiss this. It should match up with the Pecl username. It's only ment for identifying the package maintainers not the code itself.

**Review by sergeyklay — APPROVED** [review]

**Comment by niden** [comment]
Looking good bud!

**Review by niden — APPROVED** [review]

**Pull request merged**